js/tests/unit/dropdown.js

```diff
@@ -350,4 +350,30 @@ $(function () {
 
     assert.ok(!$(document.activeElement).parent().is('.disabled'), '.disabled is not focused')
   })
+
+  QUnit.test('should not close the dropdown if the user clicks on a text field/area', function (assert) {
+    assert.expect(2)
+    var dropdownHTML = '<div class="btn-group">'
+        + '<button type="button" data-toggle="dropdown">Dropdown</button>'
+        + '<ul class="dropdown-menu" role="menu">'
+        + '<li><input id="textField" type="text" /></li>'
+        + '<li><textarea id="textArea"></textarea></li>'
+        + '</ul>'
+        + '</div>'
+    var $dropdown = $(dropdownHTML)
+      .appendTo('#qunit-fixture')
+      .find('[data-toggle="dropdown"]')
+      .bootstrapDropdown()
+      .trigger('click')
+
+    $dropdown.find('#textField')
+      .trigger('click')
+
+    assert.ok($dropdown.parent('.btn-group').hasClass('open'), '.dropdown-menu is not visible')
+
+    $dropdown.find('#textArea')
```

Comment:
I think it would be good to explicitly close and re-open the dropdown before running this 2nd test.